src/lib/server/textGeneration/index.ts

```diff
@@ -61,7 +61,7 @@ async function* textGenerationWithoutTitle(
 
 	if (model.tools && !conv.assistantId) {
 		const tools = pickTools(toolsPreference, Boolean(assistant));
-		toolResults = yield* runTools(ctx, tools, preprompt);
+		if (tools.length > 0) toolResults = yield* runTools(ctx, tools, preprompt);
```

Comment:
Wouldn't `tools.length` always be 1 at a minimum ? Since [directlyAnswer](https://github.com/huggingface/chat-ui/blob/c52d8aaa958b184dab27ec1ebfe63e9f3d1de75b/src/lib/server/tools/directlyAnswer.ts#L5-L8) is always enabled and is hidden from the user. I think we'd need to filter that one out too